library/iFixit/Matryoshka/Scope.php

```diff
@@ -21,13 +21,16 @@ public function getPrefix() {
       return $this->scopePrefix ?: $this->getScopePrefix();
    }
 
-   public function getScopePrefix(bool $reset = false) {
+   public function getScopePrefix(bool $reset = false, bool $generateOnMiss = true) {
       if ($this->scopePrefix === null || $reset) {
-         $scopeValue = $this->backend->getAndSet($this->getScopeKey(),
-          function() {
-            return substr(md5(microtime() . $this->scopeName), 0, 16);
-         }, 0, $reset);
-
+         $scopeValue = $reset ? self::MISS : $this->backend->get($this->getScopeKey());
+         if ($scopeValue === self::MISS && !$generateOnMiss) {
+            return self::MISS;
+         }
+         if ($reset || ($scopeValue === self::MISS && $generateOnMiss)) {
```

Comment:
This guard seems unnecessarily complex - if `$reset` is truthy then `$scopeValue` will be MISS at this point

```suggestion
         if ($scopeValue === self::MISS && $generateOnMiss) {
```

Reply:
Good call. I refactored this several times and left it in a non-optimal state.